assets/js/pages/criarpost.js

```diff
@@ -0,0 +1,20 @@
+function PostarFoto() {
+    let f = document.getElementById("fotoPost")
+    let l = document.getElementById("legendaPost")
+    let legenda = l.value
+
+    if(f.value.trim() == '' || l.value.trim() == ''){
```

Comment:
uma outra pratica que costumamos fazer no desenvolvimento é usar nomes bem descritivos para as variaveis, de maneira que se outro desenvolvedor olhe a variavel, saiba do que se trata. Nesse caso, a variavel f e l estão um pouco vagas. Minha sugestao:

```suggestion
    let fotoPost = document.getElementById("fotoPost")
    let legendaPost = document.getElementById("legendaPost")
    let legenda = l.value
```

ou se nao for possivel utilizar o mesmo nome do elemento

```suggestion
    let fotoInput = document.getElementById("fotoPost")
    let legendaInput = document.getElementById("legendaPost")
    let legenda = l.value
```

o que acha? voce consegue revisar o restante das variaveis?